### ISSUE TITLE
the Zero-shot transfer results of CLIP and FairCLIP

### ISSUE THREAD

**Issue by 920927** [Harvard-Ophthalmology-AI-Lab/FairCLIP#7]
Hello! I followed the README in the FairCLIP folder, but the results of CLIP and FairCLIP are opposite to those in your paper. Specifically, the Zero-shot transfer results of CLIP for race attribute is 0.7004 and FairCLIP is 0.6748.  Here FairCLIP'S performance is lower than CLIP while your article is the opposite. I tryed the experiment using gender attribute and got the same conclusion.  Is there something wrong with what I'm doing?

**Comment by gcmurray** [comment]
I also have some questions related to reproducing the results from this paper.

Following the experimental setup in the supplemental material, 

> we pre-train both methods on our Harvard-FairVLMed dataset after initializing from the official checkpoints. We fine-tune CLIP for 10 epochs...

It seems like by pre-train you mean fine-tune; so, for the model fine-tuned on medical images, we fine-tune CLIP with the SLO fundus data, correct?

Then it states the hyper-parameters, and some augmentation, and then,

> Finally, all models are pre-trained on the paired fundus images and clinical notes from Harvard-FairVLMed using a batch size of 32 for 50 epochs on a single V100 GPU."

1. For FairCLIP fine-tuned on medical images: we have naturally pre-trained CLIP fine-tuned for 10 epochs without Sinkhorn loss, then 50 epochs fine-tuned with Sinkhorn loss?

2. For CLIP fine-tuned on medical images: we have naturally pre-trained CLIP fine-tuned for 10 epochs without Sinkhorn loss, and then another 50 epochs without Sinkhorn loss?

3. For FairCLIP pre-trained on natural images: we train the naturally pre-trained CLIP for 50 epochs with Sinkhorn loss?

4. For CLIP pre-trained on natural images: we train the naturally pre-trained CLIP for 50 epochs without Sinkhorn loss?

Then we apply to the zero-shot task of glaucoma prediction with "a photo of glaucoma" and "a photo of non-glaucoma"?

Finally, the code selects the best model by AUC; are you selecting the best model by AUC for 1.) selection after pre-training on 10 epochs, 2.) selection after training on 50 epochs? Or are you selecting the last checkpoint in either/both cases?

Please let me know if I've missed anything.

**Comment by luoyan407** [comment]
@920927 @gcmurray Thank you for your interest in this project. Sorry for the late reply.

I was able to reproduce the reported performance using our local database. To investigate the reproduction issue, I pulled the repository and conducted the zero-shot experiments from scratch.  

I have identified the source of the performance difference — it stems from changes in the Torch version and its compatibility with CUDA. In the initial version of FairCLIP/fairclip.yml (commit ID: 628b4acc462cf0fe93e63764aac5e733e0e6f12d), we used *torch==1.11.0+cu113* (Torch v1.11 compiled with CUDA v11.3). However, CUDA updates automatically over time, and our workstation now runs CUDA v12.7. To align with this, I upgraded Torch using the following command:  

```bash
pip install torch==2.2.2+cu121 -f https://download.pytorch.org/whl/torch_stable.html
```  

With this update, I adjusted the hyperparameter *LAMBDA* from **1e-7** to **1e-6** and set the random seed to **5681**. Under these settings, FairCLIP achieved an AUC of **0.7233** on the metric with the attribute *Race* in zero-shot experiments using a single A6000 GPU.  

I have updated the repository with these changes. Please feel free to reach out if you have any additional questions.
